pkg/runctl/run_manager.go

```diff
@@ -381,7 +401,37 @@ func (c *runManager) setupNetworkPolicyFromConfig(runNamespace string) error {
 	return nil
 }
 
-func (c *runManager) createTektonTaskRun(pipelineRun k8s.PipelineRun) error {
+func (c *runManager) volumesWithSaSecret(ctx *runContext) []corev1api.Volume {
+	var mode int32 = 420
+	return []corev1api.Volume{
+		corev1api.Volume{Name: "service-account-token",
+			VolumeSource: corev1api.VolumeSource{
+				Secret: &corev1api.SecretVolumeSource{
+					SecretName:  c.getServiceAccountSecretName(ctx),
+					DefaultMode: &mode,
+				},
+			},
+		}}
+}
+
+func (c *runManager) getServiceAccountSecretName(ctx *runContext) string {
+	if c.testing != nil {
+		if c.testing.getServiceAccountSecretNameStub != nil {
+			return c.testing.getServiceAccountSecretNameStub(ctx)
+		}
+		return ""
+	}
```

Comment:
There should be no test logic in production code. Stubbing must be like this, nothing more:

```suggestion
	if c.testing != nil && c.testing.getServiceAccountSecretNameStub != nil {
		return c.testing.getServiceAccountSecretNameStub(ctx)
	}
```

Reply:
Changed